feat: Add build with artefact upload

indigo423 opened the pull request:
Created a GitHub actions pipeline with an artifact upload to have something people can play with. The Google Protobuf was hardcoded to OSX x86_64 hardware. I've added a plugin that detects the architecture and allows it to build also on non-OSX systems.

indigo423 committed:
feat: Add build environment

indigo423 committed:
feat: Add os detection for protobuf

mvrueden commented on Makefile:
```diff
@@ -0,0 +1,34 @@
+.DEFAULT_GOAL := compile
```
-1 on `make`.

The project setup is pretty simple and there is no need for the `make` abstraction.

mvrueden commented:
FYI: The protobuf setup is temporarily. I generated the Java files manually without the maven plugin. In general it must be even more dynamic or OpenNMS should expose them properly. Hard-copy them over from OpenNMS source is not the way to go 😱

fooker commented on Makefile:
```diff
@@ -0,0 +1,34 @@
+	mvn test
+
+.PHONY: build
+build: clean
```
```suggestion
build: compile
```

indigo423 commented:
> FYI: The protobuf setup is temporarily. I generated the Java files manually without the maven plugin. In general it must be even more dynamic or OpenNMS should expose them properly. Hard-copy them over from OpenNMS source is not the way to go 😱

Thanks for the explanation, but what are the consequences of your explanation to the PR?

indigo423 committed:
chg: Don't clean by default

Consider review, to not clean all the time by default. The mvn install goal includes the "compile".
